src/value.rs

```diff
@@ -369,11 +369,7 @@ macro_rules! impl_try_truncate_into {
     (@primitive $from: ident, $into: ident, $to_primitive:path) => {
         impl TryTruncateInto<$into, TrapKind> for $from {
             fn try_truncate_into(self) -> Result<$into, TrapKind> {
-                // Casting from a float to an integer will round the float towards zero
-                num_rational::BigRational::from_float(self)
-                    .map(|val| val.to_integer())
-                    .and_then(|val| $to_primitive(&val))
-                    .ok_or(TrapKind::InvalidConversionToInt)
+                $to_primitive(&self.trunc()).ok_or(TrapKind::InvalidConversionToInt)
```

Comment:
As far I remember the current code was introduced to support 32-bit x86 platforms. Your PR is essentially reverting it here, right?